### DISCUSSION TITLE
v1.116.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#12952] [Announcements]
# v1.116.0

## Highlights

Welcome to release `v1.116.0` of Immich!

This release continues to focus on bug fixes, which is quite typical for us, especially after [large releases](https://github.com/immich-app/immich/releases/tag/v1.113.0). Highlights for this release include:

- Improved`.immich` file boot checks
- Improved external library scanning
- Timezone improvements
- Tag clean up job
- Better person merge workflow
- Improve settings pages (web)
- Fixed possible startup failure due to .immich files
- Serve `style.json` directly from `tiles.immich.app`
- New "random" api for 3rd party apps
- Notable fix: Uncaught error causes the hashing process to abort entirely in the mobile app

### Issues with .immich files on boot in 1.115.0
There were a small number of users who experienced problems with booting Immich after upgrading to 1.115.0. This was due to the new .immich file checks that verify your library folders are still mounted and writable. **If, and only if** you had problems booting 1.115.0 that persist into 1.116.0 with an error of "Unable to write .immich file, already exists", you can fix this by stopping Immich and removing the .immich files in the UPLOAD_LOCATION subfolders (upload, library, thumbnails, etc). If you experience any  further problems, please do contact us on discord or via a github discussion.

### Improved external library scanning
The external library scan mechanism has been reduced from four seperate options, to one. There is now a single button "Scan" that will handle all cases. Files that are offline (deleted or missing) now get hidden from the timeline and will show in the trash folder. This will become a dedicated screen in a future release. If the file comes back online, then the assets will re-appear in the main timeline and albums they were in previously. If the asset is offline for longer than the trash duration set in the admin config, the assets will be removed from immich automatically

### Timezone improvements
Previously, if the timezone of an asset was unknown, it would apply an arbitrary offset by assuming the time was UTC and adjusting it to the viewer's timezone. Now, a timezone will only be shown if it is known for the asset.

| Before | After |
| - | - |
| ![image](https://github.com/user-attachments/assets/f2fe666a-1946-4c43-a4eb-21449a1deb3c) | ![image](https://github.com/user-attachments/assets/d2596363-8054-4dba-936a-b916d2301aee) |
 

### Tag cleanup job
If you have tags with no assets in them, you can remove them by running the "Tag clean up" job
[here](https://my.immich.app/admin/jobs-status).

![image](https://github.com/user-attachments/assets/41986b5c-5119-420f-b642-38bb8d3b9735)

### Better person merge workflow
The merge page always uses the same direction to merge people together. The "source" gets merged into the "target". However, sometimes the target is an unnamed person. Now, in these situations, the merge direction is automatically reversed, leading to a better user experience.

### Serve `style.json` directly from `tiles.immich.app`

Up to this release, the immich server served the `style.json` file. That file includes the styling of the map as well as the font/glyphs/tiles provider. However, this also means that tile server updates were always bound to immich server updates. With this change, we decouple the two by serving the (static)`style.json` file from the infrastructure we serve the map tiles. This is also in line with pretty much any other map/tile provider. This change will assist us in bringing map improvements to all users in the future, such as map localisation without having to manage multiple different immich versions.
However, note that this does not affect the ability to specify a different `style.json` URL in the admin settings.

### UI improvement for settings pages

Each setting now has a new design accordion section with a distinguishable icon to help identify it easier as the options get longer. You can also search for the administration settings by their title and subtitle in your native language.

![image](https://github.com/user-attachments/assets/f1e4fef7-34af-44b8-b984-e3ac07ade08b)

### New "random" API
We are deprecating `GET /assets/random` in favor of `POST /search/random`, which works very similar to the other search endpoints. It includes more relations and filters. In the future, we plan to add an option to search and filter by a specific album ID.

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

### 🌟 Enhancements
* fix(web): better merge direction by @jrasm91 in https://github.com/immich-app/immich/pull/12601
* fix(server): include partner assets in random endpoint by @jrasm91 in https://github.com/immich-app/immich/pull/12599
* feat: tag cleanup job by @jrasm91 in https://github.com/immich-app/immich/pull/12654
* feat(web): improve UI/UX for settings pages by @alextran1502 in https://github.com/immich-app/immich/pull/12626
* feat(server): sort assets randomly from the API 'api/search/metadata' endpoint by including 'order': 'rand' in the API call. by @jschwalbe in https://github.com/immich-app/immich/pull/12741
* feat: serve map tile styles from tiles.immich.cloud by @zackpollard in https://github.com/immich-app/immich/pull/12858
* feat(web): fixed combobox positioning by @ben-basten in https://github.com/immich-app/immich/pull/12848
* refactor(server): library syncing by @etnoy in https://github.com/immich-app/immich/pull/12220
### 🐛 Bug fixes
* fix(web): load original panorama if specified in user settings by @Saschl in https://github.com/immich-app/immich/pull/12123
* fix(web): bucket by `localDateTime` by @jrasm91 in https://github.com/immich-app/immich/pull/12612
* fix(web): memory viewer by @jrasm91 in https://github.com/immich-app/immich/pull/12649
* fix(server): increase person search limit by @jrasm91 in https://github.com/immich-app/immich/pull/12619
* fix(web): scrollbar shows when not need by @alextran1502 in https://github.com/immich-app/immich/pull/12659
* fix(server): fix modify date extraction by @alextran1502 in https://github.com/immich-app/immich/pull/12658
* fix: empty and restore over 1,000 items by @jrasm91 in https://github.com/immich-app/immich/pull/12751
* fix(web): upgrade old style theme preference by @jrasm91 in https://github.com/immich-app/immich/pull/12775
* fix(web): asset grid with invalid asset id (at) by @jrasm91 in https://github.com/immich-app/immich/pull/12772
* fix: creating tags with leading/traling slashes by @danieldietzler in https://github.com/immich-app/immich/pull/12778
* fix(web): only show the timezone when it is known by @jrasm91 in https://github.com/immich-app/immich/pull/12779
* fix(mobile): Issue Selecting Many Albuns for Backup by @sshubhm in https://github.com/immich-app/immich/pull/12784
* fix: rework file handling so we always explicitly create, overwrite or both by @zackpollard in https://github.com/immich-app/immich/pull/12812
* fix(mobile): fix uncaught error in getting file cause hashing procses to be aborted entirely by @alextran1502 in https://github.com/immich-app/immich/pull/12826
* feat(server): validate rating by @dreadster3 in https://github.com/immich-app/immich/pull/12855
* fix(web): Fix same key for delete and stack actions by @Caesiumhydroxid in https://github.com/immich-app/immich/pull/12865
* fix: normalize external domain by @danieldietzler in https://github.com/immich-app/immich/pull/12831
* fix(server): gracefully handle unknown jobs by @jrasm91 in https://github.com/immich-app/immich/pull/12870
* fix: show asset count for unassigned faces by @danieldietzler in https://github.com/immich-app/immich/pull/12871
* fix: remove no longer needed LD_LIBRARY_PATH by @jrasm91 in https://github.com/immich-app/immich/pull/12872
* fix: set min values for face detection to reasonable values by @danieldietzler in https://github.com/immich-app/immich/pull/12877
* fix: album title state weirdness by @danieldietzler in https://github.com/immich-app/immich/pull/12874
* fix: do not use trashed assets as album covers by @jrasm91 in https://github.com/immich-app/immich/pull/12905
* fix(server): http error parsing on endpoints without a default response by @jrasm91 in https://github.com/immich-app/immich/pull/12927
* fix(web): delete non-empty album by @jrasm91 in https://github.com/immich-app/immich/pull/12937
### 📚 Documentation
* docs: scaling immich guide by @bo0tzz in https://github.com/immich-app/immich/pull/12593
* docs: add hint for healthcheck.start_interval by @kastl-ars in https://github.com/immich-app/immich/pull/12643
* docs: mobile architecture diagram by @fyfrey in https://github.com/immich-app/immich/pull/12841
* docs: add section for Traefik Reverse Proxy by @zp33dy in https://github.com/immich-app/immich/pull/12813
* fix(docs): typo in remote-access.md by @JonOcto in https://github.com/immich-app/immich/pull/12895
* docs: details for windows users how to change docker volume by @mmomjian in https://github.com/immich-app/immich/pull/12551
* docs: hidden files cursed knowledge by @jrasm91 in https://github.com/immich-app/immich/pull/12929
### 🌐 Translations
* fix(mobile): fix mn-MN.json file name  by @aryiu in https://github.com/immich-app/immich/pull/12558
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/12590
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/12737

## New Contributors
* @kastl-ars made their first contribution in https://github.com/immich-app/immich/pull/12643
* @sshubhm made their first contribution in https://github.com/immich-app/immich/pull/12784
* @zp33dy made their first contribution in https://github.com/immich-app/immich/pull/12813
* @dreadster3 made their first contribution in https://github.com/immich-app/immich/pull/12855
* @Caesiumhydroxid made their first contribution in https://github.com/immich-app/immich/pull/12865
* @jschwalbe made their first contribution in https://github.com/immich-app/immich/pull/12741
* @JonOcto made their first contribution in https://github.com/immich-app/immich/pull/12895
* @carykees98 made their first contribution in https://github.com/immich-app/immich/pull/12899

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.115.0...

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.116.0'>v1.116.0</a>.</em>

**Comment by samluxh** [comment]
Big thank you for your effort ❤️

Please add a option to be able to search for pictures of a specific year or month without the full date. Example: August 2022 or August (all years) or just 2022 (all months and days).

**Reply by SitramSoft** [reply]
That feature would be awesome

**Comment by ThatCoffeeGuy** [comment]
`immich_server            | [Nest] 17  - 09/26/2024, 8:15:43 PM     LOG [Api:StorageService] Verifying system mount folder checks (enabled=true)
immich_server            | [Nest] 17  - 09/26/2024, 8:15:43 PM   ERROR [Api:StorageService] Failed to read upload/encoded-video/.immich: Error: ENOENT: no such file or directory, open 'upload/encoded-video/.immich'
immich_server            | [Nest] 17  - 09/26/2024, 8:15:43 PM   ERROR [Api:StorageService] The "encoded-video" folder appears to be offline/missing, please make sure the volume is mounted with the correct permissions
immich_server            | api worker exited with code 1`


docker-compose relevant part:
      
`/mnt/SSD1TBVIRT/IMMICH_ENCODED_VIDEO:/usr/src/app/upload/encoded-video/`


Same issue. immich has all read write access, can touch as user and as root to the folder, I'm clueless. Everything else works.

**Reply by jrasm91** [reply]
The problem is there should be an .immich file there and there isn't one

**Reply by ThatCoffeeGuy** [reply]
Resolved, it seems I missed copying some folders and started troubleshooting the wrong issue.

**Comment by 2ipm1nd** [comment]
When will the ability to rotate photos on the web be added?

**Comment by psla** [comment]
Amazing progress as always.

I get a feeling though that the main page loads significantly slower now than it used to (on the web)

Specifically,
```
https://my-instance/api/timeline/bucket?isArchived=false&size=MONTH&timeBucket=2024-09-01T00%3A00%3A00.000Z&withPartners=true&withStacked=true
```

this request takes 500ms, and returns 1.6MB, and it *only* starts after 900ms from page load.

It looks like that request waits for another request to finish first: 

```
https://my-instance/api/timeline/buckets?isArchived=false&size=MONTH&withPartners=true&withStacked=true
```
This first request takes 550ms. So those two requests take over 1 second.

![image](https://github.com/user-attachments/assets/73cde653-27fb-424e-bf54-9d3dce36da0e)

The main page loads almost instantly, but the previews take another ~1.5 seconds. I guess you first calculate the timeline view / distribution, and then fetch the first page / correct page for the timeline?

I am wondering if something changed in 1.116, because I feel like it used to be faster. (perhaps my library increased in size, as I scanned a lot of new items). Maybe it's just an illusion that it's slower.

Also - have you ever considered something more efficient than json (e.g. protos)? It feels like there is a lot of overhead in json; although I guess it should compress nicely. 1.6MB of json is quite a bit :) 

In any case, I love immich. I have almost 400k images and videos in it.

**Reply by alextran1502** [reply]
Thanks for the report, definitely there are rooms for improvement.

Cc @midzelis and @mertalev fyi

**Reply by mertalev** [reply]
We're working on making the time bucket requests faster by fetching less data upfront and optimizing the DB queries. The one you mentioned is a bit unfortunate in that it relies on a full table scan in the DB to aggregate assets into date buckets, so it's a bigger challenge to make that one fast.

Re: JSON, I definitely agree and almost anything else would be better. msgpack seemed like an easy option last I checked since it's dynamic. Protobuf is the most efficient in terms of data, but the fact that it relies on code generation is clunky and its performance varies by language.
